### PR TITLE
send the make dirty root parent and make it dirty when a child is upd…

### PR DESCRIPTION
**Pull request by GrenadeJane**
…ated

---

This change is [Reviewable](https://reviewable.io/reviews/fishingcactus/openfl/9)

### CODE REVIEW

File: openfl/display/DisplayObjectContainer.hx [hunk 1]
```diff
@@ -51,7 +51,16 @@ class DisplayObjectContainer extends InteractiveObject {
 			
 			__children.push (child);
 			child.parent = this;
-
```
Comment (crazyjul) [inline openfl/display/DisplayObjectContainer.hx:54]: This variable needs to be updated when cacheAsBitmap is changed

Comment (GrenadeJane) [inline openfl/display/DisplayObjectContainer.hx:54]: So set dirty the direct parent ?

Comment (crazyjul) [inline openfl/display/DisplayObjectContainer.hx:54]: No, update cached_parent when cacheAsBitmap changes value

File: openfl/display/DisplayObject.hx [hunk 2]
```diff
@@ -64,6 +64,7 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable implement
 	public var name (get, set):String;
 	public var opaqueBackground:Null <Int>;
 	public var parent (default, null):DisplayObjectContainer;
+	public var cached_parent (default, null):DisplayObjectContainer;
```
Comment (crazyjul) [inline openfl/display/DisplayObject.hx:67]: wrong naming cachedParent

File: openfl/display/DisplayObject.hx [hunk 3]
```diff
@@ -1034,8 +1077,17 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable implement
 	private function set_cacheAsBitmap (value:Bool):Bool {
 		
 		__setRenderDirty ();
-		return __cacheAsBitmap = __forceCacheAsBitmap ? true : value;
-		
+
+		if (value) {
+			resetCached_parent (this);
```
Comment (crazyjul) [inline openfl/display/DisplayObject.hx:1082]: Not right, only do this if `this.cachedParent == null`

File: openfl/display/DisplayObject.hx [hunk 4]
```diff
@@ -1034,8 +1077,17 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable implement
...
+
+		} else {
+			resetCached_parent (getCached_parent(this));
```
Comment (crazyjul) [inline openfl/display/DisplayObject.hx:1085]: just use `this.cachedParent`

File: openfl/display/DisplayObject.hx [hunk 5]
```diff
@@ -242,10 +243,50 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable implement
 	public function localToGlobal (point:Point):Point {
 		
 		return __getWorldTransform ().transformPoint (point);
-		
+
 	}
-	
-	
+
+	private function getCached_parent (origin:DisplayObject) :DisplayObject {
```
Comment (crazyjul) [inline openfl/display/DisplayObject.hx:249]: Could be removed

File: openfl/display/DisplayObject.hx [hunk 6]
```diff
@@ -242,10 +243,50 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable implement
...
+				} else {
+					return getCached_parent (origin.parent);
+
+				}
+		}
+		return null;
+
+	}
+
+	private function resetCached_parent (cachedParent:DisplayObject){
```
Comment (crazyjul) [inline openfl/display/DisplayObject.hx:263]: indentation is not consistent

File: openfl/display/DisplayObject.hx [hunk 7]
```diff
@@ -693,7 +734,9 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable implement
 			__updateFilters = filters != null && filters.length > 0;
 			__renderDirty = true;
 			__worldRenderDirty++;
-			
+			if (cached_parent != null)
+				cached_parent.__setRenderDirty();
```
Comment (crazyjul) [inline openfl/display/DisplayObject.hx:738]: use `{}` for single line `if`s

File: openfl/display/DisplayObjectContainer.hx [hunk 8]
```diff
@@ -51,7 +51,16 @@ class DisplayObjectContainer extends InteractiveObject {
 			
 			__children.push (child);
 			child.parent = this;
-			
+
+			if( this.cached_parent != null)
+			{
+				child.cached_parent = this.cached_parent;
+			}
+			else if (this.__cacheAsBitmap )
```
Comment (crazyjul) [inline openfl/display/DisplayObjectContainer.hx:59]: spacing consistency all around

File: openfl/display/DisplayObjectContainer.hx [hunk 9]
```diff
@@ -51,7 +51,16 @@ class DisplayObjectContainer extends InteractiveObject {
 			
 			__children.push (child);
 			child.parent = this;
-			
+
+			if( this.cached_parent != null)
```
Comment (crazyjul) [inline openfl/display/DisplayObjectContainer.hx:55]: use coding standard for `if` : 

```
if (condition) {

} else {


} 
```

**Pull request closed without merging**